backend/src/main/java/com/capstone/moneytree/service/impl/DefaultUserService.java

```diff
@@ -406,6 +411,18 @@ public void deleteUserByEmail(String email) {
                 .collect(toList());
     }
 
+    @Override
+    public List<User> getFollowersWhoOwnsTheStock(Long id, String symbol) {
+        List<User> followersWhoOwnsThisStock = new ArrayList<>();
+
+        User user = getUserById(id);
+        Stock stock = stockService.getStockBySymbol(symbol);
+        List<Follows> followers = followsDao.findByUserToFollowId(user.getId());
+
+        followers.forEach(follower -> addFollowerToList(follower, followersWhoOwnsThisStock, stock));
+        return followersWhoOwnsThisStock;
+    }
+
```

Comment:
This one can be kept in server side since followers are not too big. We could eventually limit it too. But if the list of followers is in the thousands, it becomes slow.